hbase_sql/src/main/java/com/splicemachine/derby/lifecycle/HEngineSqlEnv.java

```diff
@@ -78,22 +83,22 @@
     private OlapClient olapClient;
     private OperationManager operationManager;
     private ZkServiceDiscovery serviceDiscovery;
-    private static final String sparkNumNodesZkPath =
-            HConfiguration.getConfiguration().getSpliceRootPath() + SPARK_NUM_NODES_PATH;
+    private static final String sparkYarnConfigZkPath =
+            HConfiguration.getConfiguration().getSpliceRootPath() + SPARK_YARN_CONFIG_PATH;
 
 
-    // Find the number of nodes on which Spark executors can be run.
-    private static int getNumSparkNodes() {
-        int numNodes = -1;
+    // Find the Spark and YARN configuration from Zookeeper.
+    private static byte [] getSparkYarnConfigBytes() {
+        byte [] returnData = null;
         try {
-            byte [] data = ZkUtils.getData(sparkNumNodesZkPath);
+            byte [] data = ZkUtils.getData(sparkYarnConfigZkPath);
```

Comment:
If it's not a lot of work make this a Protobuf message for future extendability.

Reply:
This is an ephemeral ZK field, so we don't need to be backwards compatible.  It goes away every time zookeeper reboots.  Would it matter to use protobuf?

Reply:
I created https://splicemachine.atlassian.net/browse/DB-11068 to change all messages in OlapServerMaster. It could matter for online upgrade.

Reply:
OK, we can handle it through the Jira.